### PR TITLE
feat: share game score within other apps

### PR DESCRIPTION
**Pull request by alexbrebu**
Captură de ecran din 2022-01-19 la 19 30 00

Captură de ecran din 2022-01-19 la 19 30 10

Captură de ecran din 2022-01-19 la 19 30 26

Captură de ecran din 2022-01-19 la 19 30 31

Captură de ecran din 2022-01-19 la 19 30 55

Captură de ecran din 2022-01-19 la 19 31 03

### CODE REVIEW

File: app/src/main/java/com/example/sudoku/Win.kt [hunk 1]
```diff
@@ -57,4 +62,16 @@ class Win : Fragment() {
...
+    private fun composeMessage(message: String) {
+        val share = Intent.createChooser(Intent().apply {
+            action = Intent.ACTION_SEND
+            putExtra(Intent.EXTRA_TEXT, message)
+            type="text/plain"
+            putExtra(Intent.EXTRA_TITLE, "My Sudoku Score")
+
+            flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+        }, null)
+        startActivity(share)
```
Comment (mihai-iacov) [inline app/src/main/java/com/example/sudoku/Win.kt:75]: Nice!

File: app/src/main/res/layout/fragment_win.xml [hunk 2]
```diff
@@ -42,11 +42,18 @@
             android:layout_height="wrap_content"
             android:text="@string/add_highscore_button" />
 
+        <Button
+            android:text="@string/share_score"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:id="@+id/shareScoreButton" />
```
Comment (mihai-iacov) [inline app/src/main/res/layout/fragment_win.xml]: So far, we've used the snake-case (C style) namings in the XML files, like `share_score_button` and `win_score_text`.
The Android framework is smart enough to convert the object references into camel case (Java style) in the Kotlin classes:
`binding.winScoreText`

**Review by mihai-iacov — APPROVED** [review]
Looking great!

File: app/src/main/res/layout/fragment_win.xml (hunk 2 above)
Comment (alexbrebu) [reply app/src/main/res/layout/fragment_win.xml]: Indeed. I have just updated it

File: app/src/main/java/com/example/sudoku/Win.kt (hunk 1 above)
Comment (alexbrebu) [reply app/src/main/java/com/example/sudoku/Win.kt:75]: I have just added some images to the description. Feel free to check

**Comment by vmdamian** [comment]
Looking good to me too. I'm now thinking that this button would fit better in the scoreboard, to share the score after you add it in there. But anyway, this is more of a design thing :) The functionality works great!

**Review by vmdamian — APPROVED** [review]

**Pull request merged**